hazelcast/src/main/java/com/hazelcast/scheduledexecutor/IScheduledExecutorService.java

```diff
@@ -0,0 +1,416 @@
+                                                           long delay, TimeUnit unit);
+
+    /**
+     * Creates and executes a periodic action that becomes enabled first
+     * after the given initial delay, and subsequently with the given
+     * period on all {@link Member}s given. Executions will commence after
+     * {@code initialDelay} then {@code initialDelay+period}, then
+     * {@code initialDelay + 2 * period}, and so on.
+     * If any execution of this task
+     * takes longer than its period, then subsequent execution will be skipped.
+     *
+     * <br/>
+     * <b>Note: </b> In the event of Member leaving the cluster, for whatever reason, the task is lost.
+     *
+     * @param command the task to execute
+     * @param members the collections of members - where to execute the task
+     * @param initialDelay the time to delay first execution
+     * @param period the period between successive executions
+     * @param unit the time unit of the initialDelay and period parameters
+     * @return a ScheduledFuture representing pending completion of
+     *         the task, and whose {@code get()} method will throw an
+     *         exception upon cancellation
+     * @throws RejectedExecutionException if the task cannot be
+     *         scheduled for execution
+     * @throws NullPointerException if command is null
+     */
+    Map<Member, IScheduledFuture<?>> scheduleOnMembersWithRepetition(Runnable command, Collection<Member> members,
+                                                                     long initialDelay, long period, TimeUnit unit);
+
+    /**
+     * Creates a new {@link IScheduledFuture} from the given handler.
+     * This is useful in case your member node or client from which the original
+     * scheduling happened, went down, and now you want to access the <tt>ScheduledFuture</tt> again.
+     *
+     * @param handler The handler of the task as found from {@link IScheduledFuture#getHandler()}
+     * @param <V> The return type of callable tasks
+     * @return A new {@link IScheduledFuture} from the given handler.
+     */
+    <V> IScheduledFuture<V> getScheduledFuture(ScheduledTaskHandler handler);
+
+    /**
+     * Fetches and returns all scheduled (not disposed yet) futures from all members in the cluster.
+     * If a member has no running tasks for this scheduler, it wont be included in the returned {@link Map}.
+     *
+     * @return A {@link Map} with {@link Member} keys and a List of {@link IScheduledFuture} found for this scheduler.
+     */
+    <V> Map<Member, List<IScheduledFuture<V>>> getAllScheduledFutures();
+
+    /**
+     * Initiates an orderly shutdown in which previously submitted
+     * tasks are executed, but no new tasks will be accepted.
+     * Invocation has no additional effect if already shut down.
+     *
+     * <p>This method does not wait for previously submitted tasks to
+     * complete execution.
```

Comment:
Indeed, similar to j.u.c.Executor this is an async shutdown, work keeps going on, and this doesn't block until its done. I could modify the comment to make it more clear, but I think keeping it same to j.u.c makes it more trivial in terms of consistency.